File: Arrays/Reverse_An_Array/ReverseAnArray.js
const readline = require('readline');

function reverseAnArray(){
    var start = -1;
    var end = arr.length;
    while (++start < --end) {
        [arr[start], arr[end]] = [arr[end], arr[start]]; // Swapping using es6(ES2015) Destructuring assignment
    }
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

  var arr = new Array();
  
  rl.question('Enter the number of elements in the array ', (answer) => {
    rl.on('line', (input) => {
        var nums = input.split(' ');
        for(var i = 0; i < nums.length; i++)
            arr.push(nums[i]);
        if (arr.length < answer) {
            console.log(`You have entered only ${arr.length} numbers. Please enter remaining ${answer-arr.length} numbers`);
        }
        else if (arr.length > answer) {
            console.log(`You have entered only more than ${answer} numbers. No Problem! I will reverse the entered array of numbers for you`);
            rl.close();     
        }
        else {
            rl.close();     
            reverseAnArray();
            console.log(arr.toString());
        }
    });
  });

File: Arrays/Reverse_An_Array/C/ReverseAnArray.c
#include <stdio.h>

void printArray();
void reverseAnArray();


int main(){

    int n;
    printf("Enter the size of array: ");
    scanf("%d", &n);
    int arr[n];
    for (int i = 0; i < n; i++) {
       scanf("%d", &arr[i]); 
    }
    reverseAnArray(arr, n);
    printArray(arr, n);
    return 0;
}

void printArray(int arr[], int n) {

    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }

}

void reverseAnArray(int arr[], int n) {

    int start = -1, end = n;
    while ( ++start < --end ) {
        (arr[start] ^= arr[end]), (arr[end] ^= arr[start]), (arr[start] ^= arr[end]);
    }

}